demos/common/cpp_gapi/utils_gapi/include/utils_gapi/onnx_backend.hpp

```diff
@@ -0,0 +1,120 @@
+    }
+};
+
+namespace {
+struct CPUProvider{};
+template<class ExecNetwork, class Provider = CPUProvider>
+struct ProviderApplicator {
+    static void apply(ExecNetwork &, const ModelConfig &, const inference_backends_t &backends) {
+        static_assert(std::is_same<Provider, CPUProvider>::value, "Unsupported ONNX provider requested. Please add a partial specialization if required");
+        if (backends.size() > 1) {
+            throw std::runtime_error("ONNX CPU execution provider must be only provider in the provider list or "
+                                     "be the last in order");
+        }
+
+        const BackendDescription &backend = backends.front();
+        if (backend.properties.size() >= 2) {
+            throw std::runtime_error("ONNX CPU execution provider doesn't support any arguments");
+        }
+    }
+};
+
+#ifdef GAPI_ONNX_BACKEND_EP_EXTENSION
+template<class ExecNetwork>
+struct ProviderApplicator<ExecNetwork, cv::gapi::onnx::ep::DirectML> {
+    static void apply(ExecNetwork &net, const ModelConfig &config, const inference_backends_t &backends) {
+        std::string ep_selected_device = config.deviceName;
+        const BackendDescription &backend = backends.front();
+        if (backend.properties.size() >= 2) {
+            ep_selected_device = backend.properties[1]; // second arg for ONNX is device
+        }
+        net.cfgAddExecutionProvider(cv::gapi::onnx::ep::DirectML(ep_selected_device))
+           .cfgDisableMemPattern();
+    }
+};
+
+template<class ExecNetwork>
+struct ProviderApplicator<ExecNetwork, cv::gapi::onnx::ep::OpenVINO> {
+    static void apply(ExecNetwork &net, const ModelConfig &config, const inference_backends_t &backends) {
+        std::string ep_selected_device = config.deviceName;
+        const BackendDescription &backend = backends.front();
+        if (backend.properties.size() >= 2) {
+            ep_selected_device = backend.properties[1]; // second arg for ONNX is device
+        }
+        net.cfgAddExecutionProvider(cv::gapi::onnx::ep::OpenVINO(ep_selected_device));
+    }
+};
+#endif // GAPI_ONNX_BACKEND_EP_EXTENSION
+}
+
+template<class ExecNetwork, class ...Args>
+void applyONNXProviders(ExecNetwork& net, const ModelConfig &config, inference_backends_t backend_cfgs) {
+    if (backend_cfgs.empty()) {
+        return;
+    }
+    static const std::map<std::string,
+                          std::function<void(ExecNetwork&,
+                                             const ModelConfig &,
+                                             const inference_backends_t &)>
+                         > maps {
+        {"CPU",  &ProviderApplicator<ExecNetwork>::apply},
+        {"",     &ProviderApplicator<ExecNetwork>::apply},
+#ifdef GAPI_ONNX_BACKEND_EP_EXTENSION
+        {"DML",  &ProviderApplicator<ExecNetwork, cv::gapi::onnx::ep::DirectML>::apply},
+        {"OVEP", &ProviderApplicator<ExecNetwork, cv::gapi::onnx::ep::OpenVINO>::apply}
```

Comment:
You have `EP` suffix for OV, but not for DML. Stay consistent

Reply:
ok, i'll remove it. as discussed this suffix is not important